packages/components/input/src/use-input.ts

```diff
@@ -510,8 +510,8 @@ export function useInput<T extends HTMLInputElement | HTMLTextAreaElement = HTML
     (props = {}) => {
       return {
         ...props,
-        role: "button",
-        tabIndex: 0,
+        type: "button",
```

Comment:
```suggestion
        type: "button",
        tabIndex: -1,
```

It seems better for keyboard users if the clear button is not focusable via the Tab key.

see: https://github.com/mui/material-ui/issues/30815#issuecomment-2296573576

Reply:
Thank you for the suggestion!
I've updated the code to set tabIndex to -1 as recommended.